### PR TITLE
resolve todos

### PR DESCRIPTION
**Pull request by vankop**
**What kind of change does this PR introduce?**

refactoring

**Did you add tests for your changes?**

no need

**Does this PR introduce a breaking change?**

no

**What needs to be documented once your changes are merged?**

nothing

### CODE REVIEW

File: lib/EntryPlugin.js [hunk 1]
```diff
@@ -17,12 +17,12 @@ class EntryPlugin {
 	 *
 	 * @param {string} context context path
 	 * @param {string} entry entry path
-	 * @param {EntryOptions | string=} options entry options (passing a string is deprecated)
+	 * @param {EntryOptions} options entry options
 	 */
 	constructor(context, entry, options) {
 		this.context = context;
```
Comment (sokra) [inline lib/EntryPlugin.js:23]: Throw a error here if `typeof options !== "object"` to help migrating

**Review by sokra — CHANGES_REQUESTED** [review]

File: lib/EntryPlugin.js (hunk 1 above)
Comment (vankop) [reply lib/EntryPlugin.js:27]: done

**Review by sokra — APPROVED** [review]

File: lib/index.js [hunk 2]
```diff
@@ -302,10 +302,10 @@ module.exports = mergeExports(fn, {
 	// TODO webpack 7 remove
 	get WebpackOptionsValidationError() {
 		return util.deprecate(
-			require("schema-utils").ValidationError,
+			() => require("schema-utils").ValidationError,
```
Comment (vankop) [inline lib/index.js:305]: `util.deprecate` aworks with classes

Comment (sokra) [reply lib/index.js:305]: but you only get the deprecation message when calling `new WebpackOptionsValidationError`, `instanceof WebpackOptionsValidationError` would still work fine without triggering the warning

**Comment by sokra** [comment]
Thanks

**Pull request merged**